### PR TITLE
_.defaults() does not initialize for undefined/null

### PR DESCRIPTION
**Pull request by alewiscondev**
Fixes issue https://github.com/jashkenas/underscore/issues/2115. Defaults null and undefined objects correctly.

Paired with @jridgewell and @djrconcepts!

### CODE REVIEW

**Comment by jdalton** [comment]
Instead maybe passing values through `Object(value)`. That way there can be a consistency to it.

**Comment by alewiscondev** [comment]
Good call @jdalton, fixed!

File: underscore.js [hunk 1]
```diff
@@ -1008,9 +1008,10 @@
   };
 
   // An internal function for creating assigner functions.
-  var createAssigner = function(keysFunc, undefinedOnly) {
+  var createAssigner = function(keysFunc, defaults) {
     return function(obj) {
       var length = arguments.length;
+      if (defaults) obj = Object(obj);
```
Comment (megawac) [inline underscore.js]: Any good reason we avoid this for `assign` and `extend`?

Comment (jridgewell) [inline underscore.js]: `Object.assign` throws errors for this case. Not if we want to stray too far from that.

It makes sense to me in defaults. If you have nothing, and want to default it to `{a: 1}`, `{a: 1}` should be returned.

**Pull request merged**